R/functional_structure.R

```diff
@@ -55,55 +56,46 @@ ExtractFunctionNetwork <- function(pkgName
-#'                       test coverage when creating the network representation,
-#'                       try setting this field to "test_coverage"!
+# [title] Plot Network Structure
+# [description] Plot the network structure of an R package.
+# [param] edges a data.table with SOURCE and TARGET nodes
+# [param] nodes a data.table with node and augmenting metadata
+# [param] colorFieldName (character) The name of column to use to color the nodes. This can
+#                       be any column in the "nodes" table inside the object produced by
+#                       \code{ExtractFunctionNetwork}. Default is outDegree. If you estimated
+#                       test coverage when creating the network representation,
+#                       try setting this field to "test_coverage"!
 #' @importFrom visNetwork visNetwork visHierarchicalLayout visEdges visOptions
 #' @importFrom magrittr %>%
-#' @return A plotly object (local, not on plot.ly)
-#' @export 
-#' @examples
-#' \dontrun{
-#' nw <- ExtractFunctionNetwork("lubridate")
-#' nw$nodes[,test := runif(.N)]
-#' PlotNetwork(nw, colorFieldName = "test")
-#' }
+# [return] A plotly object (local, not on plot.ly)
 PlotNetwork <- function(edges
                         , nodes
                         , colorFieldName = NULL) {
-  # format for plot
-  nodes[, id := node]
-  nodes[, label := id]
-  #nodes[, x := horizontal]
-  #nodes[, y := level]
-  
-  
-  edges[, from := SOURCE]
-  edges[, to := TARGET]
-  
-  
-  defaultNodeColor <- "blue" 
-  if (is.null(colorFieldName)) {
-    # Same Color for all nodes
-    g <- visNetwork::visNetwork(nodes = nodes, edges = edges) %>%
-        visNetwork::visHierarchicalLayout(sortMethod = "directed", direction = "DU") %>%
-        visNetwork::visEdges(arrows = 'to') %>%
-        visNetwork::visOptions(highlightNearest = list(enabled = TRUE, degree = 2000, algorithm = "hierarchical"))
+    # format for plot
+    nodes[, id := node]
+    nodes[, label := id]
+    #nodes[, x := horizontal]
+    #nodes[, y := level]
     
+    edges[, from := SOURCE]
+    edges[, to := TARGET]
     
-  } else if (is.factor(nodes[,colorFieldName, with = F]) | is.character(nodes[,colorFieldName, with = F])) {
-    # Color By Discrete Factor
-  } else {
-    # Assume continuous number scale
-  }
-  
-  print(g) # to be removed once we have an HTML report function
-  return(g)
+    defaultNodeColor <- "blue" 
+    if (is.null(colorFieldName)) {
+        
+        # Same Color for all nodes
+        g <- visNetwork::visNetwork(nodes = nodes, edges = edges) %>%
+             visNetwork::visHierarchicalLayout(sortMethod = "directed", direction = "DU") %>%
+             visNetwork::visEdges(arrows = 'to') %>%
+             visNetwork::visOptions(highlightNearest = list(enabled = TRUE, degree = 2000, algorithm = "hierarchical"))
+    
+    } else if (is.factor(nodes[,colorFieldName, with = FALSE]) | is.character(nodes[,colorFieldName, with = FALSE])) {
```

Comment:
I'll add an issue for this coloring logic.  It just needs to color the nodes based on both discrete factors (such as function type) and continuous values (such as coverage proportion).

Reply:
I know this is verbatim from `plotNetwork` function.  It's just a hanging task.

Reply:
oh nice, yeah I like using issues to track what needs to be done here. I'm sure we'll notice more stuff as we go through it